public/friendlycode/js/fc/ui/bramble-proxy.js

```diff
@@ -70,6 +71,56 @@ define(["backbone-events"], function(BackboneEvents) {
     }
   }
 
+  
+  BrambleProxy.prototype.undo = function () {
+    this.onButton("_undo");
+  };
+  BrambleProxy.prototype.redo = function () {
+    this.onButton("_redo");
+  };
+  /* This function handles all the button presses thimble has,
+   * It takes in 2 parameters, a string representing the command, and a possible option
+   * It packages a JSON object and sends it over post to thimble proxy inside brackets
+   * categoryCommand refers to what type of command should be fired inside brackets, these types being:
+   *    menuCommand: Menu Command, refers to any command outlined in the menu
+   *    viewCommand: View Command, refers to commands inside the ViewHandler
+   * command is the function that will be run within brackets
+   * params is used in conjunction with vieCommand to send extra paramters needed for viewCommand
+   */
+  BrambleProxy.prototype.onButton = function(button, option) {
+    var commandCategory = "menuCommand";
+    var command = "";
+    var params = "";
```

Comment:
Strictly speaking, this is unnecessary. Only initialize if a default value might actually be required.